login/service.go

```diff
@@ -162,18 +161,29 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.AuthorizeLoginContext, a
 	}
 
 	// First time access, redirect to oauth provider
+	redirect := ctx.Redirect
+	referrer := ctx.RequestData.Header.Get("Referer")
```

Comment:
@aslakknutsen @sbose78 OK, I added a simple whitelist which using regex. By default openshift.io is allowed only in prod mode. In dev mode, by default, any URL is valid. Can be overridden by setting up the corresponding env var.
So this PR fixes both #1073 and #892 
Please review.